File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    
    
    int maxArea(vector<int>& height) {
       
        int maxi = INT_MIN;
        int i=0;
        int j=height.size()-1;
        
        while(i<=j)
        {
            int w=0,h=0;
            
            w = abs(i-j);
            h = min(height[i],height[j]);
            
            maxi = max(maxi, w*h);
            
            if(height[i]<height[j])
            {
                i++;
            }
            else
            {
                j--;
            }
            
        }
        
        return maxi;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.go

func min(a, b int) int{
    if(a<b){
        return a
    }else {
        return b
    }
}

func max(a, b int) int{
    if(a>b){
        return a
    }else {
        return b
    }
}

func maxArea(height []int) int {
    var maxArea int = 0
    var l int = 0
    var r int = len(height)-1
    
    for l<r {
        maxArea = max(maxArea, min(height[r], height[l]) * (r-l))
        
        if height[l] < height[r]{
            l++
        }else{
            r--
        }
    }
    return maxArea
}